pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -711,23 +681,172 @@ func (r *ReconcileChaosEngine) addFinalzerToEngine(engine *chaosTypes.EngineInfo
 	return nil
 }
 
-func (r *ReconcileChaosEngine) updateStatus(engine *chaosTypes.EngineInfo, status string) error {
+func (r *ReconcileChaosEngine) updateStatus(engine *chaosTypes.EngineInfo, status litmuschaosv1alpha1.EngineStatus) error {
 	opts := client.UpdateOptions{}
 	engine.Instance.Status.EngineStatus = status
 	return r.client.Update(context.TODO(), engine.Instance, &opts)
 }
 
-func (r *ReconcileChaosEngine) updateState(engine *chaosTypes.EngineInfo, state string) error {
+func (r *ReconcileChaosEngine) updateState(engine *chaosTypes.EngineInfo, state litmuschaosv1alpha1.EngineState) error {
 	opts := client.UpdateOptions{}
 	engine.Instance.Spec.EngineState = state
 	return r.client.Update(context.TODO(), engine.Instance, &opts)
 }
 
 func checkEngineStateForStop(engine *chaosTypes.EngineInfo) bool {
 	deletetimeStamp := engine.Instance.ObjectMeta.GetDeletionTimestamp()
-	if engine.Instance.Spec.EngineState == "stop" ||
+	if engine.Instance.Spec.EngineState == litmuschaosv1alpha1.EngineStateStop ||
 		deletetimeStamp != nil {
 		return true
 	}
 	return false
 }
+
+func (r *ReconcileChaosEngine) checkRunnerPodCompleted(engine *chaosTypes.EngineInfo) bool {
+	runnerPod := corev1.Pod{}
+	r.client.Get(context.TODO(), types.NamespacedName{Name: engine.Instance.Name + "-runner", Namespace: engine.Instance.Namespace}, &runnerPod)
+	return runnerPod.Status.Phase == corev1.PodSucceeded
+}
+
+func (r *ReconcileChaosEngine) removeDefaultChaosResources(request reconcile.Request) (reconcile.Result, error) {
+	if err := r.removeChaosRunner(engine, request); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := r.removeChaosServices(engine, request); err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileChaosEngine) removeChaosRunner(engine *chaosTypes.EngineInfo, request reconcile.Request) error {
+	optsList := []client.ListOption{
+		client.InNamespace(request.NamespacedName.Namespace),
+		client.MatchingLabels{"app": engine.Instance.Name, "chaosUID": string(engine.Instance.UID)},
```

Comment:
Can we use the name of chaos-runner to identify -- label with key app is a little too generic. We run the risk of deleting retained jobs as well?